src/sources/opentelemetry/mod.rs

```diff
@@ -0,0 +1,8 @@
+mod log;
+
+use crate::config::SourceDescription;
+use log::OpentelemetryLogConfig;
+
+inventory::submit! {
+    SourceDescription::new::<OpentelemetryLogConfig>("opentelemetry")
+}
```

Comment:
Here we register an `opentelemtry` source that is wired to an `OpentelemetryLogConfig`, I think we would rather prefer an `OpentelemetryConfig` that would have placeholder to further support otel metrics (and traces) in the same source.